core/os/android/adb/perfetto.go

```diff
@@ -29,6 +29,95 @@ import (
 	"github.com/google/gapid/core/log"
 )
 
+const (
+	perfettoProducerLaunchersDirectory = "/data/local/tmp/perfetto_producer_launchers"
+)
+
+func (b *binding) preparePerfettoProducerLauncherFromApk(ctx context.Context, packageName string, launcherBinary string) (string, error) {
+	if b.Instance().GetConfiguration().GetOS().GetAPIVersion() < 29 {
+		return "", log.Errf(ctx, nil, "Querying perfetto capability requires Android API >= 29")
+	}
+	launcherPath := perfettoProducerLaunchersDirectory + "/" + launcherBinary
+	if _, err := b.Shell("rm", "-f", launcherPath).Call(ctx); err != nil {
+		return "", log.Errf(ctx, err, "Can't clean up existing perfetto producer launcher %v", launcherBinary)
+	}
+
+	// Attempt to create the directory
+	b.Shell("mkdir", "-p", perfettoProducerLaunchersDirectory).Call(ctx)
```

Comment:
Use `b.TempFile` to give you a temporary file and extract the binary directly into that file. Then delete it when we are shutting down.